engine/app/components/citizens_advice_components/button.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module CitizensAdviceComponents
+  class Button < Base
+    renders_one :icon_left
+    renders_one :icon_right
+
+    attr_reader :variant, :type, :additional_attributes
+
+    def initialize(variant: :primary, type: :button, **additional_attributes)
```

Comment:
Oh interesting re `**additional_attributes`!  Just when I thought I knew ruby.  What are the advantages of doing it this way over how we did it in inputs etc?

Reply:
The `**` kinda means the rest of the arguments has. Mostly makes the API a little simpler. 

```
= render CitizensAdviceComponents::Button.new(variant: :secondary, "data-thing": "test", "aria-label": "custom label")
```

vs.

= render CitizensAdviceComponents::Button.new(variant: :secondary, additional_attributes: { "data-thing": "test", "aria-label": "custom label"} )